src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -132,14 +138,15 @@ public static NodeSettings Default(Network network = null, ProtocolVersion proto
         }
 
         /// <summary>
-        /// Initializes configuration from command line arguments.
-        /// <para>This includes loading configuration from file.</para>
+        /// Records the command line arguments.
         /// </summary>
         /// <param name="args">Application command line arguments.</param>
-        /// <returns>Initialized node configuration.</returns>
-        /// <exception cref="ConfigurationException">Thrown in case of any problems with the configuration file or command line arguments.</exception>
-        public NodeSettings LoadArguments(string[] args)
+        /// <returns>Node configuration with arguments recorded.</returns>
+        /// <exception cref="ConfigurationException">Thrown in case of any problems with the command line arguments.</exception>
+        public NodeSettings LoadArguments(string[] args, bool delayedProcessing = false)
```

Comment:
@quantumagi I'm not really a fan of boolean overloading like this. It hides the intent of the boolean when you call the LoadArguments method from outside this class. I.e. its not obvious what the boolean is for. I prefer explicit methods so that the intent is always clear to the consumer. 

Personally, I would rename `LoadArguments` to `Load` and then create a `LoadDelayed` method which explicitly only loads the basic settings and delays the loading of the config file.

Then you can check whether the ConfigReader is null to determine if it has been loaded instead of checking a "Processed" property.

Also, this way, a feature will either call Load() or LoadDelayed() based on its requirements.

Thoughts?

Reply:
I like your suggestion of using the ConfigReader variable instead of introducing the additional `Processed` property.